### PR TITLE
Fix `SerializationTest`

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Avoids stealing leading bytes from following `SelfSerializable`'s by only using PBJ schemas and parsers on `byte[]` instead of wrapped `SerializableDataInput/OutputStream`'s.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java [hunk 1]
```diff
@@ -198,4 +205,45 @@ private static boolean isAsciiLetter(char ch) {
...
+     *
+     * @param codec a {@link Codec} to use to parse the {@code byte[]} array
+     * @param in a {@link SerializableDataInputStream} to read the array length and array from
+     * @return the result of parsing the {@code byte[]} array into an object with the codec
+     * @param <T> the type of object used by the codec
+     * @throws IOException if there is an error reading from the stream
+     */
+    public static <T> T deserializeViaBytes(
+            @NonNull final Codec<T> codec, @NonNull final SerializableDataInputStream in) throws IOException {
+        final var length = in.readInt();
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java:222]: Null checks missing

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java [hunk 2]
```diff
@@ -198,4 +205,45 @@ private static boolean isAsciiLetter(char ch) {
...
+     * @param data the object to serialize with the codec
+     * @param codec a {@link Codec} to use to serialize the object
+     * @param out a {@link SerializableDataOutputStream} to write the array length and array to
+     * @param <T> the type of object used by the codec
+     * @throws IOException if there is an error writing to the stream
+     */
+    public static <T> void serializeViaBytes(
+            @NonNull final T data, @NonNull final Codec<T> codec, @NonNull final SerializableDataOutputStream out)
+            throws IOException {
+        final var baos = new ByteArrayOutputStream();
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java:242]: baos is never closed

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java [hunk 3]
```diff
@@ -198,4 +205,45 @@ private static boolean isAsciiLetter(char ch) {
...
+     * array itself to the stream.
+     *
+     * @param data the object to serialize with the codec
+     * @param codec a {@link Codec} to use to serialize the object
+     * @param out a {@link SerializableDataOutputStream} to write the array length and array to
+     * @param <T> the type of object used by the codec
+     * @throws IOException if there is an error writing to the stream
+     */
+    public static <T> void serializeViaBytes(
+            @NonNull final T data, @NonNull final Codec<T> codec, @NonNull final SerializableDataOutputStream out)
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java:240]: null Checks missing

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/StateLifecyclesTest.java [hunk 4]
```diff
@@ -16,118 +16,151 @@
...
+    void noOpConstructorExists() {
+        assertDoesNotThrow(() -> new MerkleHederaState());
+    }
+
+    @Test
+    @Disabled
+    void testGenesisState() {
+        ClassLoaderHelper.loadClassPathDependencies();
+
+        final var merkleState = tracked(new MerkleHederaState((t, s) -> {}, (e, m, p) -> {}, (s, p, ds, t, v) -> {}));
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/StateLifecyclesTest.java:79]: can you extract the params. This line is not really readable.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/StateLifecyclesTest.java [hunk 5]
```diff
@@ -16,118 +16,151 @@
...
+
+        assertDoesNotThrow(() -> merkleState.init(platform, new DualStateImpl(), InitTrigger.GENESIS, null));
+    }
+
+    @Test
+    void testLoadingMHState() {
+        ClassLoaderHelper.loadClassPathDependencies();
+        final AtomicReference<SignedState> ref = new AtomicReference<>();
+        assertThrows(
+                com.swirlds.common.io.exceptions.ClassNotFoundException.class,
```
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/StateLifecyclesTest.java:97]: Can you import the class?

**Review by edward-swirldslabs — APPROVED** [review]
I am code reviewing the 1 line addition in the SettingConstant.java file to which platform-hashgraph team is a code-owner.  I have seen the code change, have no objections, but cannot weigh in on the validity or correctness of the rest of the merge request.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SettingConstants.java [hunk 6]
```diff
@@ -72,6 +72,7 @@ public final class SettingConstants {
     static final boolean LOAD_KEYS_FROM_PFX_FILES_DEFAULT_VALUE = true;
     static final int MAX_TRANSACTION_BYTES_PER_EVENT_DEFAULT_VALUE = 245760;
     static final int MAX_TRANSACTION_COUNT_PER_EVENT_DEFAULT_VALUE = 245760;
+    static final boolean TRANS_THROTTLE_DEFAULT_VALUE = false;
```
Comment (poulok) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SettingConstants.java:75]: `transThrottle` has been removed from the baseline

**Review by nathanklick — CHANGES_REQUESTED** [review]
Javadoc syntax errors need to be corrected as it breaks the build. Merge conflicts need resolution.

**Comment by tinker-michaelj** [comment]
Obsolete.

**Pull request closed without merging**